src/model.cpp

```diff
@@ -45,6 +45,22 @@
 
 using namespace tinyxml2;
 
+bool srdf::Model::isValidJoint(const urdf::ModelInterface& urdf_model, const std::string& name) const
+{
+  if (urdf_model.getJoint(name))
+  {
+    return true;
+  }
+  for (std::size_t k = 0; k < virtual_joints_.size(); ++k)
+  {
+    if (virtual_joints_[k].name_ == name)
+    {
+      return true;
+    }
+  }
+  return false;
```

Comment:
```suggestion
  return std::find_if(virtual_joints_.begin(), virtual_joints_.end(),
     [](const VirtualJoint& joint){ return joint.name_ == name; }) != virtual_joints_.end();
```

Reply:
Personally, I really prefer not to use this type of lambda function because I always find it unreadable. How strongly do you feel about it?

Reply:
That is one of the most convoluted ways to write a simple for-loop I have ever seen. Is there an upside to writing it this way? Surely compilers already optimize these basic for-loops?

Readable code is maintainable code, so from a maintainability perspective my gut reaction is "Oh god please no". But I'm not the strongest C++ developer, so that's just my qualified vote.

Reply:
@felixvd Which way do you find convoluted? the index-based (instead of range based) loop or the predicate-based find?
Personally, I find the first one much more readable actually, because there is no loop involved at all.

```suggestion
  auto joint_has_name = [&name](const VirtualJoint& joint){ return joint.name_ ==  name; };
  return std::find_if(virtual_joints_.cbegin(), virtual_joints_.cend(), joint_has_name);
```

Personally, I would even write
```
  auto joint_has_name { [&name](auto&& joint){ return joint.name_ ==  name; } };
  return std::find_if(virtual_joints_.cbegin(), virtual_joints_.cend(), joint_has_name);
```
or (c++20)
```
  auto joint_has_name { [&name](auto&& joint){ return joint.name_ ==  name; } };
  return std::ranges::find_if(virtual_joints_, joint_has_name);
```
But uniform initialization, universal references. or even c++20 might be a bit too much for this codebase...

Reply:
+1 not using lambdas. As this is largely a matter of taste, what if we went with original author's preference (who put the work in) rather than bike shedding?

Reply:
@DLu I'm sorry I opened this Pandora's box. I don't feel too strongly about. I find the lambda version more readable, but I don't think your version is hard to read either.

Reply:
I have settled on a slight compromise, and ditched the index variable at the suggestion of @griswaldbrooks.

Reply:
> @felixvd Which way do you find convoluted? the index-based (instead of range based) loop or the predicate-based find?

I meant the lambda. I love range-based loops, so I'm all for @DLu's current (updated) proposal.

Your alternatives are pretty and make the lambda a lot clearer than the single-line version, but any Programming 101 student understands the loop-based search, so I'll vote for that.

PS: I had only seen lambdas in other languages and wasn't familiar with the C++ syntax, so I am playing the student's role here :)

Reply:
@davetcoleman technically you encouraged bike shedding because you voted against the majority in this thread. :clown_face: 

> but any Programming 101 student understands the loop-based search, so I'll vote for that.

Simplicity is not the same as simple syntax. :smile: 

The current range-based loop is fine and this is resolved.